src/conn/conn_sweep.c

```diff
@@ -359,8 +366,14 @@ __wt_sweep_create(WT_SESSION_IMPL *session)
 	/*
 	 * Handle sweep does enough I/O it may be called upon to perform slow
 	 * operations for the block manager.
+	 *
+	 * The sweep thread sweeps the lookaside table for outdated records,
+	 * it gets its own cursor for that purpose.
+	 *
+	 * Don't tap the sweep thread for eviction.
 	 */
-	F_SET(session, WT_SESSION_CAN_WAIT);
+	F_SET(session, WT_SESSION_CAN_WAIT |
+	    WT_SESSION_LOOKASIDE_CURSOR | WT_SESSION_NO_EVICTION);
```

Comment:
> The sweep thread does eviction of dead trees and trees that become idle. What's the case where it can't do eviction?

When it's using its lookaside table cursor, it can't be grabbed for opportunistic eviction, it could deadlock.

`WT_SESSION_NO_EVICTION` is maybe a bad name, since it doesn't mean "don't evict", it means "don't re-purpose this thread to do other eviction work just because it's waiting on something".

I didn't see a reason to rope the sweep thread into doing opportunistic eviction, so I turned it off in all cases; if there's a reason the sweep thread should do opportunistic eviction, I could toggle the flag when the lookaside table cursor is in use, as I do for other threads (for example, eviction worker threads).